app/grandchallenge/archives/views.py

```diff
@@ -334,6 +358,153 @@ def get_context_data(self, **kwargs):
 
 
+class ArchiveEditArchiveItem(
+    UserFormKwargsMixin,
+    LoginRequiredMixin,
+    ObjectPermissionRequiredMixin,
+    FormView,
+):
+    form_class = ArchiveItemForm
+    template_name = "archives/archive_item_form.html"
+    permission_required = (
+        f"{Archive._meta.app_label}.upload_{Archive._meta.model_name}"
+    )
+    raise_exception = True
+
+    def get_permission_object(self):
+        return self.archive
+
+    @cached_property
+    def archive(self):
+        return get_object_or_404(Archive, slug=self.kwargs["slug"])
+
+    @cached_property
+    def archive_item(self):
+        return get_object_or_404(ArchiveItem, pk=self.kwargs["id"])
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs.update(
+            {"archive": self.archive, "archive_item": self.archive_item}
+        )
+        return kwargs
+
+    def get_context_data(self, *args, **kwargs):
+        context = super().get_context_data(*args, **kwargs)
+        context.update({"archive": self.archive})
+        return context
+
+    def form_valid(self, form):  # noqa: C901
+        def create_upload(image_files):
+            raw_files = []
+            upload_session = RawImageUploadSession.objects.create(
+                creator=self.request.user
+            )
+            for image_file in image_files:
+                raw_files.append(
+                    RawImageFile(
+                        upload_session=upload_session,
+                        filename=image_file.name,
+                        staged_file_id=image_file.uuid,
+                    )
+                )
+            RawImageFile.objects.bulk_create(list(raw_files))
+            return upload_session.pk
+
+        upload_pks = {}
+        civ_pks = set(self.archive_item.values.values_list("pk", flat=True))
+
+        for slug, value in form.cleaned_data.items():
+            if value is None:
+                continue
+            ci = ComponentInterface.objects.get(slug=slug)
+            civ = self.archive_item.values.filter(interface=ci).first()
+            if civ and civ.value is not None:
+                if civ.value == value:
+                    continue
+                civ_pks.remove(civ.pk)
+                civ = None
+            if not civ:
+                civ = ComponentInterfaceValue.objects.create(interface=ci)
+            if ci.kind in InterfaceKind.interface_type_image():
+                civ_pks.add(civ.pk)
+                upload_pks[civ.pk] = create_upload(value)
+            elif ci.kind in InterfaceKind.interface_type_file():
+                name = get_valid_filename(value[0].name)
+                with value[0].open() as f:
+                    civ.file = File(f, name=name)
+                    civ.save()
+                civ_pks.add(civ.pk)
+            else:
+                civ.value = value
+                civ.save()
+                civ_pks.add(civ.pk)
+
+        tasks = add_values_to_archive_item.signature(
```

Comment:
Is there a need to do this in a celery task? Downside is that these changes are not instant.

Reply:
Yes, because this can only be done after the image imports have finished. It could theoretically be done immediately if no image type interfaces are used.

Reply:
Clear!